lib/src/client.dart

```diff
@@ -1820,20 +1860,20 @@ class Client extends MatrixApi {
       Object? syncError;
       await _checkSyncFilter();
 
+      // The timeout we send to the server for the sync loop. It says to the
+      // server that we want to receive an empty sync response after this
+      // amount of time if nothing happens.
+      timeout ??= const Duration(seconds: 30);
+
       // Call onSoftLogout 5 minutes before access token expires to prevent
       // failing network requests.
       final tokenExpiresAt = accessTokenExpiresAt;
       if (onSoftLogout != null &&
           tokenExpiresAt != null &&
-          tokenExpiresAt.difference(DateTime.now()) <= Duration(minutes: 5)) {
-        await onSoftLogout?.call(this);
+          tokenExpiresAt.difference(DateTime.now()) <= timeout * 2) {
+        await _handleSoftLogout();
```

Comment:
If this function fails, no exception is thrown. This means we did call clear(), but try to continue with the sync. Should _handleSoftLogout throw some exception or otherwise return, that you are not signed in anymore?

Reply:
I added a `rethrow`